pycbc/waveform/bank.py

```diff
@@ -676,11 +680,36 @@ def __getitem__(self, index):
 
         # Get the waveform filter
         distance = 1.0 / DYN_RANGE_FAC
-        htilde = pycbc.waveform.get_waveform_filter(
-            tempout[0:self.filter_length], self.table[index],
-            approximant=approximant, f_lower=f_low, f_final=f_end,
-            delta_f=self.delta_f, delta_t=self.delta_t, distance=distance,
-            **self.extra_args)
+        if self.compressed_waveforms is not None :
```

Comment:
This section is complicated but self-contained. Can you move this to a method that we call instead?